shell/platform/linux/fl_clipping_view.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_CLIPPING_VIEW_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_CLIPPING_VIEW_H_
+
+#include <gtk/gtk.h>
+
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlClippingView,
+                     fl_clipping_view,
+                     FL,
+                     CLIPPING_VIEW,
+                     GtkEventBox)
+
+/**
+ * FlClippingView:
+ *
+ * #FlClippingView is a GTK widget that applies clipping mutations set by
+ * Framework side to child platform view.
+ */
+
+/**
+ * fl_clipping_view_new:
+ *
+ * Creates a new #FlClippingView widget.
+ *
+ * Returns: the newly created #FlClippingView widget.
+ */
+GtkWidget* fl_clipping_view_new();
+
+/**
+ * fl_clipping_view_reset:
+ * @clipping_view: an #FlClippingView.
+ * @child: (transfer-none): widget to apply mutators.
```

Comment:
The annotation should be "transfer none" (i.e. no hyphen). But it's not needed, as this is the default behaviour.